tests/unittests/sources/test_smartos.py

```diff
@@ -798,249 +789,300 @@ def read(self, size=-1):
-        self.assertEqual(1, written_line.count(b"\n"))
+        assert 1 == written_line.count(b"\n")
 
-    def _get_written_line(self, key="some_key"):
-        client = self._get_client()
-        client.get(key)
-        return self.serial.write.call_args[0][0]
+    def test_get_metadata_writes_bytes(self, joyent_client, m_serial):
+        assert isinstance(_get_written_line(joyent_client, m_serial), bytes)
 
-    def test_get_metadata_writes_bytes(self):
-        self.assertIsInstance(self._get_written_line(), bytes)
-
-    def test_get_metadata_line_starts_with_v2(self):
-        foo = self._get_written_line()
+    def test_get_metadata_line_starts_with_v2(self, joyent_client, m_serial):
+        foo = _get_written_line(joyent_client, m_serial)
         self.assertStartsWith(foo.decode("ascii"), b"V2".decode("ascii"))
 
-    def test_get_metadata_uses_get_command(self):
-        parts = self._get_written_line().decode("ascii").strip().split(" ")
-        self.assertEqual("GET", parts[4])
+    def test_get_metadata_uses_get_command(self, joyent_client, m_serial):
+        parts = (
+            _get_written_line(joyent_client, m_serial)
+            .decode("ascii")
+            .strip()
+            .split(" ")
+        )
+        assert "GET" == parts[4]
 
-    def test_get_metadata_base64_encodes_argument(self):
+    def test_get_metadata_base64_encodes_argument(
+        self, joyent_client, m_serial
+    ):
         key = "my_key"
-        parts = self._get_written_line(key).decode("ascii").strip().split(" ")
-        self.assertEqual(b64e(key), parts[5])
+        parts = (
+            _get_written_line(joyent_client, m_serial, key)
+            .decode("ascii")
+            .strip()
+            .split(" ")
+        )
+        assert b64e(key) == parts[5]
 
-    def test_get_metadata_calculates_length_correctly(self):
-        parts = self._get_written_line().decode("ascii").strip().split(" ")
+    def test_get_metadata_calculates_length_correctly(
+        self, joyent_client, m_serial
+    ):
+        parts = (
+            _get_written_line(joyent_client, m_serial)
+            .decode("ascii")
+            .strip()
+            .split(" ")
+        )
         expected_length = len(" ".join(parts[3:]))
-        self.assertEqual(expected_length, int(parts[1]))
+        assert expected_length == int(parts[1])
 
-    def test_get_metadata_uses_appropriate_request_id(self):
-        parts = self._get_written_line().decode("ascii").strip().split(" ")
+    def test_get_metadata_uses_appropriate_request_id(
+        self, joyent_client, m_serial
+    ):
+        parts = (
+            _get_written_line(joyent_client, m_serial)
+            .decode("ascii")
+            .strip()
+            .split(" ")
+        )
         request_id = parts[3]
-        self.assertEqual(8, len(request_id))
-        self.assertEqual(request_id, request_id.lower())
+        assert 8 == len(request_id)
+        assert request_id == request_id.lower()
```

Comment:
I don't think we need this test do we? As `test_get_metadata_uses_random_number_for_request_id` just below asserts that request_id formatted as an 8 character lowercase hex string.